detox/android/detox/src/main/java/com/wix/detox/espresso/DetoxAssertion.java

```diff
@@ -48,25 +48,21 @@ public static ViewInteraction assertNotExists(ViewInteraction i) {
     }
 
     public static void waitForAssertMatcher(final ViewInteraction i, final Matcher<View> m, double timeoutSeconds) {
-        long originTime = System.nanoTime();
-        long currentTime;
+        final long originTime = System.nanoTime();
 
         while (true) {
-            currentTime = System.nanoTime();
+            long currentTime = System.nanoTime();
             long elapsed = currentTime - originTime;
-            double seconds = (double)elapsed / 1000000000.0;
+            double seconds = (double) elapsed / 1000000000.0;
             if (seconds >= timeoutSeconds) {
-                return;
+                throw new DetoxRuntimeException("" + timeoutSeconds + "sec timeout expired without matching of given matcher: " + m);
```

Comment:
@LeoNatan Need to also do this for iOS. Ack?

Reply:
Fixed. #1473